CONTRIBUTING.md

```diff
@@ -173,3 +207,124 @@ locations.
     └─ ...
  └─ ...
 ```
+
+## PHP internals
+
+For information on PHP internal C functions see
+[PHP at the Core: A Hacker's Guide](https://php.net/internals), though this is
+considered incomplete. Various external resources can be found on the web. A
+standard printed reference is the book "Extending and Embedding PHP" by Sara
+Golemon.
+
+## Submitting patches
+
+GitHub pull requests are the preferred way to propose PHP patch. If you are not
+using GitHub, attach your patch to a PHP bug and consider sending a notification
+email about the change to internals@lists.php.net. If the bug is for an
+extension, also CC the extension maintainer. Explain what has been changed by
+your patch. Test scripts should be included.
+
+Please make the mail subject prefix `[PATCH]`. If attaching a patch, ensure it
+has a file extension of `.txt`. This is because only MIME attachments of type
+`text/*` are accepted.
+
+If you are fixing broken functionality in PHP C source code first create a bug
+or identify an existing bug at [bugs.php.net](https://bugs.php.net). A bug can
+be used to track the patch progress and prevent your changes getting lost in the
+PHP mail archives.
+
+### PHP documentation patches
```

Comment:
This section is kindof outdated in that people are expected to submit doc patches via edit.php.net right now (though hopefully that will change...)